src/main/java/org/folio/note/NoteServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package org.folio.note;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.folio.common.OkapiParams;
+import org.folio.rest.exceptions.InputValidationException;
+import org.folio.rest.jaxrs.model.Link;
+import org.folio.rest.jaxrs.model.Note;
+import org.folio.rest.jaxrs.model.NoteCollection;
+import org.folio.rest.jaxrs.model.UserDisplayInfo;
+import org.folio.userlookup.UserLookUp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.Json;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+@Component
+public class NoteServiceImpl implements NoteService {
+
+  private final Logger logger = LoggerFactory.getLogger(NoteServiceImpl.class);
+
+  @Autowired
+  private NoteRepository repository;
+
+  @Override
+  public Future<NoteCollection> getNotes(String cqlQuery, int offset, int limit, String tenantId) {
+    return repository.getNotes(cqlQuery, offset, limit, tenantId);
+  }
+
+  @Override
+  public Future<Note> addNote(Note note, OkapiParams okapiParams) {
+    return UserLookUp.getUserInfo(okapiParams.getHeadersAsMap())
+      .compose(creatorUser -> {
+        final List<Link> links = note.getLinks();
```

Comment:
validation should go before any actions: in this case before retrieving user info